tests/entities/test_run.py

```diff
@@ -35,9 +35,9 @@ def test_creation_and_hydration(self):
                             "lifecycle_stage": lifecycle_stage,
                             "artifact_uri": artifact_uri,
                             },
-                   "data": {"metrics": [dict(m) for m in metrics],
-                            "params": [dict(p) for p in params],
-                            "tags": [dict(t) for t in tags]}}
+                   "data": {"metrics": {m.key: dict(m) for m in metrics.values()},
```

Comment:
This is unrelated to this PR, but I wonder, do we call to_dictionary outside of tests? Should we just compare directly without the indirection?

Reply:
Good question, converting to dict helps us bypass the fact that MLflow entities with the same properties aren't actually considered equal (https://github.com/mlflow/mlflow/pull/961/files is an attempt to address this). We only use `to_dictionary` in one place outside tests - in the runs CLI ([link](https://github.com/mlflow/mlflow/blob/f1629c2/mlflow/runs.py#L83)).

Reply:
Created https://github.com/mlflow/mlflow/issues/1089 to track removing the to_dictionary, from_dictionary APIs for MLflow 1.0

Reply:
awesome, thanks Sid!